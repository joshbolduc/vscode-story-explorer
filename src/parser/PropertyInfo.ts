export interface PropertyInfo {
  value: unknown;
  isPartOfDeclaration: boolean;
}
