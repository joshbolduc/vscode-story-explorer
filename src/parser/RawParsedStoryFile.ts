import type { Location } from '../types/Location';

export interface RawParsedStoriesMeta {
  id: string | undefined;
  title: string | undefined;
  location: Location | undefined;
}

export interface RawParsedStory {
  location: Location | undefined;
  name: string | undefined;
  nameForId: string | undefined;
}

export interface RawParsedStoryFile {
  meta: RawParsedStoriesMeta;
  stories: RawParsedStory[];
}
