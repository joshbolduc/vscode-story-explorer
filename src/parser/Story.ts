import type { Location } from '../types/Location';

export interface Story {
  location: Location | undefined;
  name: string;
  nameForId: string | undefined;
}
