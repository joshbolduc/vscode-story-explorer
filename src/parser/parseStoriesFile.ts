import { tryParseCsf } from './csf/csf';
import { tryParseMdx } from './mdx/mdx';

const csfParser = { parser: tryParseCsf, type: 'csf' } as const;
const mdxParser = { parser: tryParseMdx, type: 'mdx' } as const;

const getParsers = (filePath?: string) => {
  if (filePath?.endsWith('.mdx') || filePath?.endsWith('.md')) {
    return [mdxParser];
  }

  return [csfParser];
};

export const parseStoriesFile = (contents: string, filePath?: string) => {
  const parsers = getParsers(filePath);

  for (const { parser, type } of parsers) {
    const parsed = parser(contents);
    if (
      parsed &&
      // .mdx files don't require a Meta tag or stories, but .stories.mdx files
      // need at least one
      (type !== 'mdx' ||
        parsed.meta.location !== undefined ||
        parsed.stories.length > 0 ||
        !filePath?.endsWith('.stories.mdx'))
    ) {
      return { ...parsed, type };
    }
  }

  return undefined;
};
