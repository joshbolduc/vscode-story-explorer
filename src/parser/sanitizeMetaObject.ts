import { sanitize } from '@componentdriven/csf';

export const sanitizeMetaObject = (
  metaProperties: Record<string, unknown> = {},
) => {
  const { id, title } = metaProperties;
  const titleAsString = typeof title === 'string' ? title : undefined;
  const idAsString = typeof id === 'string' ? id : undefined;

  const sanitizedId =
    idAsString !== undefined ? sanitize(idAsString) : undefined;

  return {
    id: sanitizedId,
    title: titleAsString,
  };
};
