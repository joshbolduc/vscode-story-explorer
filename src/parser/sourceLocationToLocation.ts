import type { SourceLocation } from '@babel/types';

export const sourceLocationToLocation = (
  sourceLocation: SourceLocation | null | undefined,
) =>
  sourceLocation
    ? {
        start: {
          line: sourceLocation.start.line,
          column: sourceLocation.start.column,
        },
        end: {
          line: sourceLocation.end.line,
          column: sourceLocation.end.column,
        },
      }
    : undefined;
