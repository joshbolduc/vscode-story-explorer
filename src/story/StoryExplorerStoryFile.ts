import { sanitize } from '@componentdriven/csf';
import startCase from 'lodash/startCase';
import type { GlobSpecifier } from '../config/GlobSpecifier';
import type { ParsedStoryWithFileUri } from '../parser/parseStoriesFileByUri';
import { StoryExplorerStory } from './StoryExplorerStory';

export class StoryExplorerStoryFile {
  public readonly docsStory?: StoryExplorerStory;

  private readonly stories: StoryExplorerStory[];

  public constructor(
    private readonly parsed: ParsedStoryWithFileUri,
    private readonly specifiers: GlobSpecifier[],
  ) {
    this.stories = parsed.stories.reduce<StoryExplorerStory[]>((acc, story) => {
      const { nameForId, location, name } = story;

      if (name) {
        const seStory = StoryExplorerStory.fromStory(
          { nameForId, location, name },
          this,
        );
        if (seStory) {
          acc.push(seStory);
        }
      }

      return acc;
    }, []);

    this.docsStory = StoryExplorerStory.fromStoryFileForDocs(this);
  }

  public getUri() {
    return this.parsed.file;
  }

  public getTitle() {
    const title = this.parsed.meta.title;

    if (title) {
      return title;
    }

    const generateTitle = (specifier: GlobSpecifier) => {
      const uriPath = this.getUri().path;
      if (!uriPath.startsWith(specifier.directory)) {
        return undefined;
      }

      const fileSuffix = uriPath.slice(specifier.directory.length);

      const extensionIndex = fileSuffix.indexOf('.');

      const suffixWithoutExtension =
        extensionIndex > 0 ? fileSuffix.slice(0, extensionIndex) : fileSuffix;

      const autoTitleParts = [
        ...(specifier.titlePrefix?.split('/') ?? []),
        ...suffixWithoutExtension.split('/'),
      ]
        .map(startCase)
        .filter(Boolean);

      return autoTitleParts.join('/');
    };

    for (const specifier of this.specifiers) {
      const autoTitle = generateTitle(specifier);

      if (autoTitle) {
        return autoTitle;
      }
    }
  }

  public getId() {
    const metaId = this.parsed.meta.id;
    if (metaId) {
      return metaId;
    }

    const title = this.getTitle();

    if (title) {
      return sanitize(title);
    }
  }

  public getStories() {
    return this.stories;
  }

  public getAllStories() {
    return [this.docsStory, ...this.stories].filter(
      Boolean,
    ) as StoryExplorerStory[];
  }

  public isDocsOnly() {
    return this.isMdx() && this.parsed.stories.length === 0;
  }

  public getMetaLocation() {
    return this.parsed.meta.location;
  }

  public isMdx() {
    return this.parsed.type === 'mdx';
  }
}
