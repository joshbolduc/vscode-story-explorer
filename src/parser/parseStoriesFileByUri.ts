import { firstValueFrom } from 'rxjs';
import type { Uri } from 'vscode';
import { supportLooseStoryNameSemantics } from '../config/supportLegacyStoryNameProperty';
import { logDebug } from '../log/log';
import { getFileContent } from '../util/getFileContent';
import { parseStoriesFile } from './parseStoriesFile';

export type ParsedStoryWithFileUri = NonNullable<
  ReturnType<typeof parseStoriesFile>
> & {
  file: Uri;
};

export const parseStoriesFileByUri = async (
  storiesFile: Uri,
): Promise<ParsedStoryWithFileUri | undefined> => {
  const contents = await getFileContent(storiesFile);

  const useLooseStoryNameSemantics = await firstValueFrom(
    supportLooseStoryNameSemantics,
  );

  const parsed = parseStoriesFile(
    contents,
    { useLooseStoryNameSemantics },
    storiesFile.path,
  );

  if (!parsed) {
    // Most likely due to transient syntax error
    logDebug('Failed to parse file', storiesFile);
    return undefined;
  }

  return { ...parsed, file: storiesFile };
};
