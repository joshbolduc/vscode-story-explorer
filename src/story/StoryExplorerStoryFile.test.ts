import { readFileSync } from 'fs';
import { basename, join } from 'path';
import { sync } from 'fast-glob';
import type { Uri } from 'vscode';
import { testBaseDir } from '../../test/util/testBaseDir';
import { parseStoriesFile } from '../parser/parseStoriesFile';
import type { ParsedStoryWithFileUri } from '../parser/parseStoriesFileByUri';
import { StoryExplorerStoryFile } from './StoryExplorerStoryFile';

describe('StoryExplorerStoryFile', () => {
  const tests = sync(['project/src/**/*.stories.*'], {
    cwd: testBaseDir,
  })
    .map((file) => {
      const storiesPath = join(testBaseDir, file);
      const contents = readFileSync(storiesPath).toString();
      const parsedRaw = parseStoriesFile(contents, basename(file));

      return { file, parsedRaw };
    })
    .filter(({ parsedRaw }) => !!parsedRaw);

  tests.forEach(({ file, parsedRaw }) => {
    it(`parses story file ${file}`, () => {
      const parsed: ParsedStoryWithFileUri = {
        ...parsedRaw!,
        file: {
          path: `/mock/basedir/${file}`,
        } as Uri,
      };

      const storyFile = new StoryExplorerStoryFile(parsed, [
        {
          directory: '/mock/basedir/project/src',
          files: '**/*',
          titlePrefix: 'Auto-generated title prefix',
        },
      ]);

      const stories = storyFile
        .getAllStories()
        .map(({ id, name, isDocs }) => ({ id, name, isDocs }));

      expect({
        id: storyFile.getId(),
        title: storyFile.getTitle(),
        isDocsOnly: storyFile.isDocsOnly(),
        stories,
      }).toMatchSnapshot();
    });
  });
});
