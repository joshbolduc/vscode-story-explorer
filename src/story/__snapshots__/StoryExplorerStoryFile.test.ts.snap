// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryExplorerStoryFile parses story file project/src/noTitle/csf/EmptyDefaultExport.stories.js 1`] = `
Object {
  "id": "auto-generated-title-prefix-no-title-csf-empty-default-export",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "auto-generated-title-prefix-no-title-csf-empty-default-export",
      "isDocs": true,
      "name": "Empty Default Export",
    },
    Object {
      "id": "auto-generated-title-prefix-no-title-csf-empty-default-export--primary",
      "isDocs": false,
      "name": "Primary",
    },
  ],
  "title": "Auto Generated Title Prefix/No Title/Csf/Empty Default Export",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/noTitle/csf/nested/WithoutTitle.stories.js 1`] = `
Object {
  "id": "auto-generated-title-prefix-no-title-csf-nested-without-title",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "auto-generated-title-prefix-no-title-csf-nested-without-title",
      "isDocs": true,
      "name": "Without Title",
    },
    Object {
      "id": "auto-generated-title-prefix-no-title-csf-nested-without-title--primary",
      "isDocs": false,
      "name": "Primary",
    },
  ],
  "title": "Auto Generated Title Prefix/No Title/Csf/Nested/Without Title",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/noTitle/mdx/MdxMeta.stories.mdx 1`] = `
Object {
  "id": "auto-generated-title-prefix-no-title-mdx-mdx-meta",
  "isDocsOnly": true,
  "stories": Array [
    Object {
      "id": "auto-generated-title-prefix-no-title-mdx-mdx-meta--page",
      "isDocs": true,
      "name": "Mdx Meta",
    },
  ],
  "title": "Auto Generated Title Prefix/No Title/Mdx/Mdx Meta",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/noTitle/mdx/MdxMetaAndStory.stories.mdx 1`] = `
Object {
  "id": "auto-generated-title-prefix-no-title-mdx-mdx-meta-and-story",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "auto-generated-title-prefix-no-title-mdx-mdx-meta-and-story",
      "isDocs": true,
      "name": "Mdx Meta And Story",
    },
    Object {
      "id": "auto-generated-title-prefix-no-title-mdx-mdx-meta-and-story--primary",
      "isDocs": false,
      "name": "Primary",
    },
  ],
  "title": "Auto Generated Title Prefix/No Title/Mdx/Mdx Meta And Story",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/noTitle/mdx/MdxStory.stories.mdx 1`] = `
Object {
  "id": "auto-generated-title-prefix-no-title-mdx-mdx-story",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "auto-generated-title-prefix-no-title-mdx-mdx-story",
      "isDocs": true,
      "name": "Mdx Story",
    },
    Object {
      "id": "auto-generated-title-prefix-no-title-mdx-mdx-story--primary",
      "isDocs": false,
      "name": "Primary",
    },
  ],
  "title": "Auto Generated Title Prefix/No Title/Mdx/Mdx Story",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/BasicCsf.stories.js 1`] = `
Object {
  "id": "example-basic-csf",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-basic-csf",
      "isDocs": true,
      "name": "Basic CSF",
    },
    Object {
      "id": "example-basic-csf--primary",
      "isDocs": false,
      "name": "Primary",
    },
    Object {
      "id": "example-basic-csf--secondary",
      "isDocs": false,
      "name": "Secondary",
    },
    Object {
      "id": "example-basic-csf--large",
      "isDocs": false,
      "name": "Large",
    },
    Object {
      "id": "example-basic-csf--small",
      "isDocs": false,
      "name": "Small",
    },
  ],
  "title": "Example/Basic CSF",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/CsfV1.stories.js 1`] = `
Object {
  "id": "example-csf-v1",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-csf-v1",
      "isDocs": true,
      "name": "CSF v1",
    },
    Object {
      "id": "example-csf-v1--v-1-story-name",
      "isDocs": false,
      "name": "Renamed via non-hoisted story",
    },
    Object {
      "id": "example-csf-v1--v-2-v-1-story-name",
      "isDocs": false,
      "name": "v2 name override",
    },
    Object {
      "id": "example-csf-v1--v-2-v-1-story-name-empty",
      "isDocs": false,
      "name": "V 2 V 1 Story Name Empty",
    },
  ],
  "title": "Example/CSF v1",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/CsfWithMdx.stories.mdx 1`] = `
Object {
  "id": "example-mdx-importing-csf",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-mdx-importing-csf",
      "isDocs": true,
      "name": "MDX importing CSF",
    },
    Object {
      "id": "example-mdx-importing-csf--basic",
      "isDocs": false,
      "name": "Basic",
    },
    Object {
      "id": "example-mdx-importing-csf--basic-2",
      "isDocs": false,
      "name": "Story name from MDX",
    },
    Object {
      "id": "example-mdx-importing-csf--basic-3",
      "isDocs": false,
      "name": "Basic 3",
    },
  ],
  "title": "Example/MDX importing CSF",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/CustomId.stories.js 1`] = `
Object {
  "id": "manually-specified-csf-story-id",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "manually-specified-csf-story-id",
      "isDocs": true,
      "name": "Custom ID CSF",
    },
    Object {
      "id": "manually-specified-csf-story-id--story",
      "isDocs": false,
      "name": "Story",
    },
  ],
  "title": "Example/Custom ID CSF",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/CustomId.stories.mdx 1`] = `
Object {
  "id": "manually-specified-mdx-id",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "manually-specified-mdx-id",
      "isDocs": true,
      "name": "Custom ID MDX",
    },
    Object {
      "id": "manually-specified-mdx-id--story",
      "isDocs": false,
      "name": "Story",
    },
  ],
  "title": "Example/Custom ID MDX",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/CustomName.stories.js 1`] = `
Object {
  "id": "example-custom-name",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-custom-name",
      "isDocs": true,
      "name": "Custom Name",
    },
    Object {
      "id": "example-custom-name--custom-name",
      "isDocs": false,
      "name": "Custom story name",
    },
    Object {
      "id": "example-custom-name--overridden",
      "isDocs": false,
      "name": "Custom story name override",
    },
    Object {
      "id": "example-custom-name--named-via-variable",
      "isDocs": false,
      "name": "Custom story name var",
    },
    Object {
      "id": "example-custom-name--interpolated-name",
      "isDocs": false,
      "name": "Custom story name var in template string",
    },
    Object {
      "id": "example-custom-name--empty-name",
      "isDocs": false,
      "name": "Empty Name",
    },
    Object {
      "id": "example-custom-name--leading-whitespace",
      "isDocs": false,
      "name": "     Leading whitespace",
    },
  ],
  "title": "Example/Custom Name",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/DeepTree.stories.js 1`] = `
Object {
  "id": "example-deep-tree-with-several-kinds",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-deep-tree-with-several-kinds",
      "isDocs": true,
      "name": "Kinds",
    },
    Object {
      "id": "example-deep-tree-with-several-kinds--story",
      "isDocs": false,
      "name": "Story",
    },
  ],
  "title": "Example/Deep/Tree/With/Several/Kinds",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/DocsOnly.stories.mdx 1`] = `
Object {
  "id": "example-mdx-docs-only",
  "isDocsOnly": true,
  "stories": Array [
    Object {
      "id": "example-mdx-docs-only--page",
      "isDocs": true,
      "name": "MDX Docs Only",
    },
  ],
  "title": "Example/MDX Docs Only",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/DocsOnlyTop.stories.mdx 1`] = `
Object {
  "id": "mdx-docs-at-top-level",
  "isDocsOnly": true,
  "stories": Array [
    Object {
      "id": "mdx-docs-at-top-level--page",
      "isDocs": true,
      "name": "MDX Docs at top level",
    },
  ],
  "title": "MDX Docs at top level",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/Excluded.stories.js 1`] = `
Object {
  "id": "test-exclusions-excluded",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "test-exclusions-excluded",
      "isDocs": true,
      "name": "Excluded",
    },
    Object {
      "id": "test-exclusions-excluded--other-export",
      "isDocs": false,
      "name": "Other Export",
    },
  ],
  "title": "Test/Exclusions/Excluded",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/ExportStyles.stories.js 1`] = `
Object {
  "id": "example-export-styles",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-export-styles",
      "isDocs": true,
      "name": "Export Styles",
    },
    Object {
      "id": "example-export-styles--normal-export",
      "isDocs": false,
      "name": "Normal Export",
    },
    Object {
      "id": "example-export-styles--alternative-export-name",
      "isDocs": false,
      "name": "Alternative Export Name",
    },
    Object {
      "id": "example-export-styles--export-name-overridden",
      "isDocs": false,
      "name": "Override name 3",
    },
  ],
  "title": "Example/Export Styles",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/Form.stories.js 1`] = `
Object {
  "id": "example-form",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-form",
      "isDocs": true,
      "name": "Form",
    },
    Object {
      "id": "example-form--form",
      "isDocs": false,
      "name": "Form",
    },
  ],
  "title": "Example/Form",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/Hoisting.stories.js 1`] = `
Object {
  "id": "example-hoisting",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-hoisting",
      "isDocs": true,
      "name": "Hoisting",
    },
    Object {
      "id": "example-hoisting--hoisting",
      "isDocs": false,
      "name": "Hoisting",
    },
  ],
  "title": "Example/Hoisting",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/Included.stories.js 1`] = `
Object {
  "id": "test-exclusions-included",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "test-exclusions-included",
      "isDocs": true,
      "name": "Included",
    },
    Object {
      "id": "test-exclusions-included--included-export",
      "isDocs": false,
      "name": "Included Export",
    },
  ],
  "title": "Test/Exclusions/Included",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/JsxExtension.stories.jsx 1`] = `
Object {
  "id": "example-jsx-extension",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-jsx-extension",
      "isDocs": true,
      "name": "JSX extension",
    },
    Object {
      "id": "example-jsx-extension--story",
      "isDocs": false,
      "name": "Story",
    },
  ],
  "title": "Example/JSX extension",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/MdxWithStories.stories.mdx 1`] = `
Object {
  "id": "example-mdx-with-stories",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-mdx-with-stories",
      "isDocs": true,
      "name": "MDX with Stories",
    },
    Object {
      "id": "example-mdx-with-stories--mdx-first-story",
      "isDocs": false,
      "name": "MDX first story",
    },
    Object {
      "id": "example-mdx-with-stories--mdx-second-story",
      "isDocs": false,
      "name": "MDX second story",
    },
    Object {
      "id": "example-mdx-with-stories--mdx-story-3",
      "isDocs": false,
      "name": "MDX Story 3",
    },
  ],
  "title": "Example/MDX with Stories",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/MdxWithStoriesTop.stories.mdx 1`] = `
Object {
  "id": "mdx-with-stories-top",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "mdx-with-stories-top",
      "isDocs": true,
      "name": "MDX with stories, top",
    },
    Object {
      "id": "mdx-with-stories-top--mdx-with-stories-top",
      "isDocs": false,
      "name": "MDX with stories, top",
    },
  ],
  "title": "MDX with stories, top",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/MergeOne.stories.js 1`] = `
Object {
  "id": "example-merge",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-merge",
      "isDocs": true,
      "name": "Merge",
    },
    Object {
      "id": "example-merge--one",
      "isDocs": false,
      "name": "One",
    },
  ],
  "title": "Example/Merge",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/MergeTwo.stories.js 1`] = `
Object {
  "id": "example-merge",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-merge",
      "isDocs": true,
      "name": "Merge",
    },
    Object {
      "id": "example-merge--two",
      "isDocs": false,
      "name": "Two",
    },
  ],
  "title": "Example/Merge",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/Precedence.stories.mdx 1`] = `
Object {
  "id": "example-precedence",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-precedence",
      "isDocs": true,
      "name": "Precedence",
    },
    Object {
      "id": "example-precedence--small",
      "isDocs": false,
      "name": "Story with name, story, body",
    },
    Object {
      "id": "example-precedence--story-with-name-body",
      "isDocs": false,
      "name": "Story with name, body",
    },
  ],
  "title": "Example/Precedence",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/SameName.stories.js 1`] = `
Object {
  "id": "example-with-same-name",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-with-same-name",
      "isDocs": true,
      "name": "With Same Name",
    },
    Object {
      "id": "example-with-same-name--same-name",
      "isDocs": false,
      "name": "Same Name",
    },
  ],
  "title": "Example/With Same Name",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/SameNameChild.stories.js 1`] = `
Object {
  "id": "example-with-same-name-same-name",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-with-same-name-same-name",
      "isDocs": true,
      "name": "Same Name",
    },
    Object {
      "id": "example-with-same-name-same-name--nested-story",
      "isDocs": false,
      "name": "Nested Story",
    },
  ],
  "title": "Example/With Same Name/Same Name",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/StoryDeclarations.stories.js 1`] = `
Object {
  "id": "test-story-declarations",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "test-story-declarations",
      "isDocs": true,
      "name": "Story Declarations",
    },
    Object {
      "id": "test-story-declarations--const-declarator",
      "isDocs": false,
      "name": "Const Declarator",
    },
    Object {
      "id": "test-story-declarations--separate-declarator",
      "isDocs": false,
      "name": "Separate Declarator",
    },
    Object {
      "id": "test-story-declarations--separate-declarator-2",
      "isDocs": false,
      "name": "Separate declarator with story name",
    },
    Object {
      "id": "test-story-declarations--multiple-one",
      "isDocs": false,
      "name": "Multiple One",
    },
    Object {
      "id": "test-story-declarations--multiple-two",
      "isDocs": false,
      "name": "Multiple two (given name)",
    },
    Object {
      "id": "test-story-declarations--function-story",
      "isDocs": false,
      "name": "Renamed Function Story",
    },
    Object {
      "id": "test-story-declarations--separate-function-story",
      "isDocs": false,
      "name": "Separate Function Story",
    },
    Object {
      "id": "test-story-declarations--separate-function-story-2",
      "isDocs": false,
      "name": "Separate function story with story name",
    },
  ],
  "title": "Test/Story Declarations",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/TitleInterpolation.stories.js 1`] = `
Object {
  "id": "example-title-interpolation",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "example-title-interpolation",
      "isDocs": true,
      "name": "Title Interpolation",
    },
    Object {
      "id": "example-title-interpolation--primary",
      "isDocs": false,
      "name": "Primary",
    },
  ],
  "title": "Example/Title Interpolation",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/Top.stories.js 1`] = `
Object {
  "id": "top",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "top",
      "isDocs": true,
      "name": "Top",
    },
    Object {
      "id": "top--top-level-story",
      "isDocs": false,
      "name": "Top Level Story",
    },
  ],
  "title": "Top",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/TopHoisted.stories.js 1`] = `
Object {
  "id": "top-hoisted",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "top-hoisted",
      "isDocs": true,
      "name": "Top Hoisted",
    },
    Object {
      "id": "top-hoisted--top-hoisted",
      "isDocs": false,
      "name": "Top Hoisted",
    },
  ],
  "title": "Top Hoisted",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/TypeScript.stories.tsx 1`] = `
Object {
  "id": "test-typescript-sample",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "test-typescript-sample",
      "isDocs": true,
      "name": "Sample",
    },
    Object {
      "id": "test-typescript-sample--type-script-story",
      "isDocs": false,
      "name": "Type Script Story",
    },
    Object {
      "id": "test-typescript-sample--type-script-typed-story",
      "isDocs": false,
      "name": "Type Script Typed Story",
    },
    Object {
      "id": "test-typescript-sample--type-script-typed-generic-story",
      "isDocs": false,
      "name": "Type Script Typed Generic Story",
    },
    Object {
      "id": "test-typescript-sample--type-script-type-assertion-story",
      "isDocs": false,
      "name": "Type Script Type Assertion Story",
    },
  ],
  "title": "Test/TypeScript/Sample",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/TypeScriptAsMeta.stories.ts 1`] = `
Object {
  "id": "test-typescript-default-as-meta",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "test-typescript-default-as-meta",
      "isDocs": true,
      "name": "Default as Meta",
    },
    Object {
      "id": "test-typescript-default-as-meta--type-script-story",
      "isDocs": false,
      "name": "Type Script Story",
    },
  ],
  "title": "Test/TypeScript/Default as Meta",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/TypeScriptIdentifier.stories.ts 1`] = `
Object {
  "id": "test-typescript-typed-identifier",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "test-typescript-typed-identifier",
      "isDocs": true,
      "name": "Typed Identifier",
    },
    Object {
      "id": "test-typescript-typed-identifier--type-script-story",
      "isDocs": false,
      "name": "Type Script Story",
    },
  ],
  "title": "Test/TypeScript/Typed Identifier",
}
`;

exports[`StoryExplorerStoryFile parses story file project/src/stories/TypeScriptIdentifierAsMeta.stories.ts 1`] = `
Object {
  "id": "test-typescript-identifier-exported-as-meta",
  "isDocsOnly": false,
  "stories": Array [
    Object {
      "id": "test-typescript-identifier-exported-as-meta",
      "isDocs": true,
      "name": "Identifier Exported as Meta",
    },
    Object {
      "id": "test-typescript-identifier-exported-as-meta--type-script-story",
      "isDocs": false,
      "name": "Type Script Story",
    },
  ],
  "title": "Test/TypeScript/Identifier Exported as Meta",
}
`;
